sled-storage/src/manager.rs

```diff
@@ -1159,10 +1243,23 @@ impl StorageManager {
                         requested: config.generation,
                         current: ledger_data.generation,
                     });
-                }
+                } else if config.generation == ledger_data.generation {
+                    info!(
+                        log,
+                        "Requested generation number matches prior request",
+                    );
 
-                // TODO: If the generation is equal, check that the values are
-                // also equal.
+                    if ledger_data != &config {
+                        error!(
+                            log,
+                            "Requested configuration changed (with the same generation)";
+                            "generation" => ?config.generation
+                        );
+                        return Err(Error::PhysicalDiskConfigurationChanged {
+                            generation: config.generation,
+                        });
+                    }
+                }
 
                 info!(log, "Request looks newer than prior requests");
```

Comment:
Should this be inside a final `} else { ... }` to avoid being logged in the case where the request is not newer? Do we also want an `info!()` in the "request generation matches" case since we're going to proceed with the ensure operation?

Reply:
Sure thing, updated with new logs